packages/compiler/core/checker.ts

```diff
@@ -4710,6 +4793,25 @@ export function createChecker(program: Program): Checker {
     return [false, [createUnassignableDiagnostic(source, target, diagnosticTarget)]];
   }
 
+  function isAssignableToValueType(
+    source: Type | ValueType,
+    target: ValueType,
+    diagnosticTarget: DiagnosticTarget
+  ): [boolean, readonly Diagnostic[]] {
+    if (source.kind === "Value") {
+      return isTypeAssignableTo(source.target, target.target, diagnosticTarget);
+    }
+    const [assignable, diagnostics] = isTypeAssignableTo(source, target.target, diagnosticTarget);
+    if (!assignable) {
+      return [assignable, diagnostics];
+    }
+
+    if (!isValueType(source)) {
```

Comment:
Is this preventing keyed model literals from being assignable via `valueof `?  (i.e. `{a: "my string"}` not being assignable to `valueof {a: string}`)